tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+        data = response.data
+        self.assertEquals(self.get_hashed_contents(), data["hexhash"])
+        self.assertEquals(tmp_name, data["filename"])
+        self.assertEquals("File contents not shown.", data["contents"])
+        self.assertEquals(file_url(self.get_hashed_contents()), data["url"])
+        return response
+
+    def test_reject_duplicates(self):
+        self.test_create_file()
+        with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+            tmp.write(self.file_contents.encode("utf-8"))
+            tmp.seek(0)
+            response = self.client.post(
+                "/rp/job/upload/", {"file": tmp})
+        self.assertEquals(400, response.status_code)
+        self.assertIn("error", response.data)
+        self.assertEquals("File already present.", response.data["error"])
+
+    def test_reject_large(self):
+        with patch("regparser.web.jobs.views.FileUploadView",
+                   new=PatchedFileUploadView) as p:
+            p.size_limit = 10000
+            with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+                tmp.write(self.file_contents.encode("utf-8"))
+                tmp.seek(0)
+                response = self.client.post(
+                    "/rp/job/upload/", {"file": tmp})
+            self.assertEquals(201, response.status_code)
+
+            with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+                contents = "123" * 100001
+                tmp.write(contents.encode("utf-8"))
+                tmp.seek(0)
+                response = self.client.post(
+                    "/rp/job/upload/", {"file": tmp})
+            self.assertEquals(400, response.status_code)
+            self.assertEquals("File too large (10000-byte limit).",
+                              response.data["error"])
+
+    def test_create_and_read_and_delete(self):
+        expected = self.test_create_file().data
+        url = urlparse(expected["url"])
+        response = self.client.get(url.path)
+        contents = response.content.decode("utf-8")
+        self.assertEquals(self.file_contents, contents)
+
+        response = self.client.get("/rp/job/upload/", format="json")
+        self.assertEquals(1, len(response.data))
+        data = response.data[0]
+        self.assertEquals("File contents not shown.", data["contents"])
+        self.assertEquals(expected["file"], data["file"])
+        self.assertEquals(expected["filename"], data["filename"])
+        self.assertEquals(self.get_hashed_contents(), data["hexhash"])
+        self.assertEquals(url.path, urlparse(data["url"]).path)
+
+        response = self.client.delete(url.path)
+        self.assertEqual(204, response.status_code)
+
+        response = self.client.get(url.path)
+        self.assertEqual(404, response.status_code)
+
+        response = self.client.get("/rp/job/upload/", format="json")
+        data = response.data
+        self.assertEquals(0, len(data))
```

Comment:
It is: it's the only test of GET against `/rp/job/upload/`, the only test of DELETE for the file endpoint, and of what happens when you try to GET file after deletion. (Deleting a ProposalPipelineJob that refers to an uploaded file doesn't try to delete that file, because that would be bad for the ability to re-use files.)